client/forest-manager/src/prove.rs

```diff
@@ -44,6 +44,10 @@ where
     let next = iter.next().transpose()?;
     let prev = iter.next_back().transpose()?;
 
+    // This is just done to allow the `apply_delta` function to remove files from the trie without failing because of an incomplete database.
+    iter.next().transpose()?;
+    iter.next_back().transpose()?;
```

Comment:
Wouldn't it be more efficient to handle this in the `Scenario 1` case?

Reply:
Yep, changed.